packages/babel-traverse/src/path/introspection.ts

```diff
@@ -396,7 +399,10 @@ export function _guessExecutionStatusRelativeToDifferentFunctions(
   // then we can be a bit smarter and handle cases where the function is either
   // a. not called at all (part of an export)
   // b. called directly
-  const binding = target.scope.getBinding(target.node.id.name);
+  const bindingName = targetIsArrowExpressionDeclaration
+    ? target.container.id.name
```

Comment:
```suggestion
    ? target.parent.id.name
```

Not in this specific case, but in general `container` could be an array. Btw, `make tscheck` will run TS on this file.